src/sing/stateMachine/sequencerStateMachine.ts

```diff
@@ -505,6 +497,169 @@ class MoveNoteState implements IState<State, Input, Context> {
   }
 }
 
+class ResizeNoteLeftState implements IState<State, Input, Context> {
+  readonly id = "resizeNoteLeft";
+
+  private readonly cursorPosAtStart: PositionOnSequencer;
+  private readonly targetTrackId: TrackId;
+  private readonly targetNoteIds: Set<NoteId>;
+  private readonly mouseDownNoteId: NoteId;
+
+  private currentCursorPos: PositionOnSequencer;
+
+  private innerContext:
+    | {
+        targetNotesAtStart: Map<NoteId, Note>;
+        previewRequestId: number;
+        executePreviewProcess: boolean;
+        edited: boolean;
+        guideLineTicksAtStart: number;
+      }
+    | undefined;
+
+  constructor(
+    cursorPosAtStart: PositionOnSequencer,
+    targetTrackId: TrackId,
+    targetNoteIds: Set<NoteId>,
+    mouseDownNoteId: NoteId,
+  ) {
+    if (!targetNoteIds.has(mouseDownNoteId)) {
+      throw new Error("mouseDownNoteId is not included in targetNoteIds.");
+    }
+    this.cursorPosAtStart = cursorPosAtStart;
+    this.targetTrackId = targetTrackId;
+    this.targetNoteIds = targetNoteIds;
+    this.mouseDownNoteId = mouseDownNoteId;
+
+    this.currentCursorPos = cursorPosAtStart;
+  }
+
+  private previewResizeLeft(context: Context) {
+    if (this.innerContext == undefined) {
+      throw new Error("innerContext is undefined.");
+    }
+    const snapTicks = context.snapTicks.value;
+    const previewNotes = context.previewNotes.value;
+    const targetNotesAtStart = this.innerContext.targetNotesAtStart;
+    const mouseDownNote = getOrThrow(targetNotesAtStart, this.mouseDownNoteId);
+    const dragTicks = this.currentCursorPos.ticks - this.cursorPosAtStart.ticks;
+    const notePos = mouseDownNote.position;
+    const newNotePos =
+      Math.round((notePos + dragTicks) / snapTicks) * snapTicks;
+    const movingTicks = newNotePos - notePos;
+
+    const editedNotes = new Map<NoteId, Note>();
+    for (const note of previewNotes) {
+      const targetNoteAtStart = getOrThrow(targetNotesAtStart, note.id);
+      const notePos = targetNoteAtStart.position;
+      const noteEndPos =
+        targetNoteAtStart.position + targetNoteAtStart.duration;
+      const position = Math.min(noteEndPos - snapTicks, notePos + movingTicks);
+      const duration = noteEndPos - position;
+
+      if (note.position !== position && note.duration !== duration) {
```

Comment:
（間違ってるかもなのですが）
`||`かも･･･？
```suggestion
      if (note.position !== position || note.duration !== duration) {
```